tests/integration/helpers/loginByGoogle.js

```diff
@@ -0,0 +1,45 @@
+import { ClientFunction, Role } from "testcafe";
+const fetch = require("node-fetch");
+
+const setWindowAccessToken = ClientFunction((data) =>
+  window.__store__.dispatch("auth/setAccessToken", data)
+);
+
+export const googleAuthUser = Role(
+  `${process.env.BROWSERSTACK_BASE_URL}/login`,
+  async (t) => {
+    await fetch("https://www.googleapis.com/oauth2/v4/token", {
+      method: "POST",
+      body: JSON.stringify({
+        grant_type: "refresh_token",
+        client_id: process.env.GOOGLE_OAUTH2_CLIENT_ID,
+        client_secret: process.env.GOOGLE_OAUTH2_CLIENT_SECRET,
+        refresh_token: process.env.GOOGLE_OAUTH2_REFRESH_TOKEN,
+      }),
+      headers: { "Content-Type": "application/json" },
+    })
+      .then((res) => res.json())
+      .then(async (data) => {
+        const { access_token: socialAuthToken } = data;
+
+        await fetch(process.env.DJANGO_CONVERT_SOCIAL_AUTH_TOKEN_URL, {
+          method: "POST",
+          body: JSON.stringify({
+            grant_type: "convert_token",
+            client_id: process.env.DJANGO_DEFAULT_OAUTH2_CLIENT_ID,
+            client_secret: process.env.DJANGO_DEFAULT_OAUTH2_CLIENT_SECRET,
+            backend: "google-oauth2",
+            token: socialAuthToken,
+          }),
+          headers: { "Content-Type": "application/json" },
+        })
+          .then((res) => res.json())
+          .then(
+            async (data) => {
+              await setWindowAccessToken(data);
+            },
+            () => {}
+          );
```

Comment:
can't we use clicks on the interface to do instead instead of writing all the calls again manually?

Reply:
The only call manually written is for Google Login. I was using the normal Google login before but it had limitations:
1. I was using my gmail to login. Would not prefer it due to security issues. I thought that we can create a gmail account reserved for this, but
2. When the CI tried logging in with my account, I got an alert from gmail to verify this login as I was signing in from different location. The test case failed as it became a 2-step verification.

Another limitations which I can guess:
1. There will still be security issues if we use a gmail account.
2. There may be some limitations on sign-ins for these activities, which may result in blocking from Gmail. Also, as the test cases grow, these sign-ins will only multiply.

Once the user is logged-in, there is no other customization done.